frontend/src/pages/GraphPF/TrafficAnimation/TrafficPointRendererPF.tsx

```diff
@@ -0,0 +1,178 @@
+import * as React from 'react';
+import { kialiStyle } from 'styles/StyleUtils';
+import { keyframes } from 'typestyle';
+import { Edge } from '@patternfly/react-topology';
+
+export abstract class TrafficPointRenderer {
+  abstract render(
+    element: Edge,
+    animationDelay: string,
+    isInfinite: boolean,
+    onAnimationEnd?: React.AnimationEventHandler
+  ): React.SVGProps<SVGElement>;
+}
+
+function getMoveAnimation(edge: Edge, percentVisible: number, isInfinite: boolean): string {
+  const startPoint = edge.getStartPoint();
+  const endPoint = edge.getEndPoint();
+  const moveAnimation = {};
+
+  if (edge.getBendpoints().length === 0) {
+    const moveX = endPoint.x - startPoint.x;
+    const moveY = endPoint.y - startPoint.y;
+
+    moveAnimation['0%'] = { translate: '0' };
+    moveAnimation[`${percentVisible}%`] = { translate: `${moveX}px ${moveY}px`, display: isInfinite ? '' : 'none' };
```

Comment:
The circle and diamond should appear in the graph when they start moving.
```suggestion
    moveAnimation['0%'] = { opacity: 1, translate: '0' };
    moveAnimation[`${percentVisible}%`] = { opacity: 1, translate: `${moveX}px ${moveY}px`, display: isInfinite ? '' : 'none' };
```